protocol-correctness/lib/functions/list-no-common-item.k

```diff
@@ -0,0 +1,21 @@
+require "protocol-correctness/lib/language/base-types.k"
+require "protocol-correctness/lib/language/expression.k"
+
+module LIST-NO-COMMON-ITEM-SYNTAX
+  imports BASE-TYPES-SYNTAX
+  imports BOOL
+  imports MAP
+
+  syntax Bool ::= listNoCommonItem(Usize, Map, ExpressionList)  [function, functional]
+endmodule
+
+module LIST-NO-COMMON-ITEM
+  imports LIST-NO-COMMON-ITEM-SYNTAX
+
+  imports EXPRESSION-SYNTAX
+
+  rule listNoCommonItem(_:Usize, _:Map, [.]) => true
```

Comment:
What does the first argument `USize` do?

Reply:
I added documentation, but the short story is that this function is used in a loop's invariant, and, in the recursive call, it attempts to produce a map identical to the one produced by the loop, which makes it easy to apply the claim co-inductively.

If I would reimplement this, I would probably remove the usize argument and I would apply a function that transforms all the map keys to 0 or something like that.